fc/mef.py

```diff
@@ -217,265 +52,152 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
     Notes
     -----
     GMM finds clusters by fitting a linear combination of `n_clusters`
-    Gaussian probability density functions (pdf) to `data`, by likelihood
-    maximization.
+    Gaussian probability density functions (pdf) to `data` using
+    Expectation Maximization (EM).
+
+    GMM can be fairly sensitive to the initial parameter choice. To
+    generate a reasonable set of initial conditions, `clustering_gmm`
+    first divides all samples in `data` into `n_clusters` groups of the
+    same size based on their Euclidean distance to the origin. Then, for
+    each group, the 50% samples farther away from the mean are discarded.
+    The mean and covariance are calculated from the remaining samples of
+    each group, and used as initial conditions for the GMM EM algorithm.
 
     `clustering_gmm` internally uses `GMM` from the ``scikit-learn``
-    library. For more information, consult their documentation.
+    library, with full covariance matrices for each cluster and a fixed,
+    uniform set of weights. This means that `clustering_gmm` implicitly
+    assumes that all bead subpopulations have roughly the same number of
+    events. For more information, consult ``scikit-learn``'s documentation.
```

Comment:
Both using 50% of the samples for each cluster and forcing the weights to be equal are what's currently making GMM work. This was tested with a bunch of beads files taken by almost everybody in lab.

K-means initialization is actually performed by default (check line 486 in https://github.com/scikit-learn/scikit-learn/blob/c957249/sklearn/mixture/gmm.py). And we saw that this didn't work properly.

Using equal weights is actually less of a hack that you would think. It's a well supported, documented option in scikit's GMM. And the model I'm using with full covariance matrices is arguably less restrictive than the default, which uses equal, diagonal covariance matrices (http://scikit-learn.org/stable/modules/generated/sklearn.mixture.GMM.html).